batchproducer/batchproducer.go

```diff
@@ -36,6 +36,9 @@ type Producer interface {
 	// when this interval elapses. There’s no guarantee, however, since the main goroutine is
 	// used to send the stats and therefore there may be some skew.
 	SetStatInterval(time.Duration) error
+
+	// SetProblemPolicy sets the ProblemPolicy.
+	SetProblemPolicy(ProblemPolicy) error
```

Comment:
I think I'd rather see a `Config` object, passed on initialization, than a bunch of pre-run setters (i.e. that stop working as soon as you start the component.)